### ISSUE TITLE
# Venice Boat Classification using CNN

### ISSUE THREAD

**Issue by Anasbinmalik01** [Anasbinmalik01/Anasbinmalik01#2]
# Venice Boat Classification using CNN
# Run this in Google Colab

# Step 1: Import libraries
import tensorflow as tf
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Conv2D, MaxPooling2D, Flatten, Dense, Dropout
from tensorflow.keras.preprocessing.image import ImageDataGenerator

# Step 2: Data Preprocessing
# Assuming your dataset is in folders: train/ and test/
train_dir = "/content/train"
test_dir = "/content/test"

train_datagen = ImageDataGenerator(rescale=1./255, shear_range=0.2,
                                   zoom_range=0.2, horizontal_flip=True)

test_datagen = ImageDataGenerator(rescale=1./255)

train_set = train_datagen.flow_from_directory(
    train_dir, target_size=(128, 128), batch_size=32, class_mode='categorical')

test_set = test_datagen.flow_from_directory(
    test_dir, target_size=(128, 128), batch_size=32, class_mode='categorical')

# Step 3: Build CNN Model
model = Sequential([
    Conv2D(32, (3,3), activation='relu', input_shape=(128,128,3)),
    MaxPooling2D(pool_size=(2,2)),
    
    Conv2D(64, (3,3), activation='relu'),
    MaxPooling2D(pool_size=(2,2)),
    
    Flatten(),
    Dense(128, activation='relu'),
    Dropout(0.5),
    Dense(train_set.num_classes, activation='softmax')  # output layer
])

model.compile(optimizer='adam', loss='categorical_crossentropy', metrics=['accuracy'])

# Step 4: Train Model
history = model.fit(train_set, validation_data=test_set, epochs=10)

# Step 5: Evaluate Model
loss, accuracy = model.evaluate(test_set)
print(f"Test Accuracy: {accuracy*100:.2f}%")
